src/function/scalar/map/map.cpp

```diff
@@ -2,11 +2,69 @@
 #include "duckdb/common/string_util.hpp"
 #include "duckdb/parser/expression/bound_expression.hpp"
 #include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/function/list_aggregate_function.hpp"
+#include "duckdb/function/aggregate/nested_functions.hpp"
 #include "duckdb/common/types/data_chunk.hpp"
 #include "duckdb/common/pair.hpp"
+#include "duckdb/common/types/value_map.hpp"
 
 namespace duckdb {
 
+bool KeyListIsEmpty(list_entry_t *data, idx_t rows) {
+	for (idx_t i = 0; i < rows; i++) {
+		auto size = data[i].length;
+		if (size != 0) {
+			return false;
+		}
+	}
+	return true;
+}
+
+static bool AreKeysNull(Vector &keys, idx_t row_count) {
+	auto list_type = keys.GetType();
+	D_ASSERT(list_type.id() == LogicalTypeId::LIST);
+	auto key_type = ListType::GetChildType(list_type).id();
+	auto arg_data = ListVector::GetData(keys);
+	auto &entries = ListVector::GetEntry(keys);
+	if (key_type == LogicalTypeId::SQLNULL) {
+		if (KeyListIsEmpty(arg_data, row_count)) {
+			return false;
+		}
+		// The entire key list is NULL for one (or more) of the rows: (ARRAY[NULL, NULL, NULL])
+		return true;
+	}
+
+	VectorData list_data;
+	keys.Orrify(row_count, list_data);
+	auto validity = FlatVector::Validity(entries);
+	return (!validity.CheckAllValid(row_count));
```

Comment:
What you probably want to call is `VectorOperations::HasNull`, which accounts for different vector types